### PR TITLE
Add `docker compose wait`

### PR DESCRIPTION
**Pull request by ulyssessouza**
**What I did**
Add `wait` command

**Related issue**
Resolves https://github.com/docker/compose/issues/10395

**(not mandatory) A picture of a cute animal, if possible in relation to what you did**
sloth

### CODE REVIEW

**Review by nicksieger — APPROVED** [review]
LGTM!

**Comment by nicksieger** [comment]
One more thing, what do you think about adding a e2e or cuke test?

File: pkg/compose/wait.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	containers, err := s.getContainers(ctx, project.Name, oneOffInclude, false, options.Services...)
+	if err != nil {
+		return 0, err
+	}
+
+	var returnErrorChan = make(chan error)
+	var statusCode int64
+	for _, c := range containers {
+		c := c
+		go func() {
```
Comment (ndeloof) [inline pkg/compose/wait.go:37]: could use an errorGroup (like we do in other places in compose) to capture first error and stop other API calls by cancelable context

**Comment by ndeloof** [comment]
this new command deserves an e2e tests :)

**Comment by ulyssessouza** [comment]
@ndeloof @nicksieger  The idea was to validate the concept first (that's the reason the draft). I'll write the tests now.

File: cmd/compose/wait.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+			opts.services = services
+			statusCode, err = runWait(ctx, backend, project, &opts)
+			return err
+		}),
+		PostRun: func(cmd *cobra.Command, args []string) {
+			os.Exit(int(statusCode))
+		},
+	}
+
+	cmd.Flags().BoolVar(&opts.downProject, "down-project", false, "Drops project when the first container stops")
```
Comment (ulyssessouza) [inline cmd/compose/wait.go:54]: For now I just added this `--down-project`. But I can imagine having a `stop-project` or `stop-services`, only stopping the services we enumerated in the `wait`

**Review by ndeloof — APPROVED** [review]

**Pull request merged**